src/main/kotlin/no/nav/k9/los/integrasjon/abac/PepClient.kt

```diff
@@ -268,50 +172,69 @@ class PepClient constructor(
     }
 
     override suspend fun harTilgangTilOppgave(oppgave: Oppgave): Boolean {
-        return !oppgave.harFagSaksNummer() || harTilgangTilLesSak(
-            fagsakNummer = oppgave.fagsakSaksnummer,
-            aktørid = oppgave.aktorId
-        )
+        return harTilgang("k9sak", Action.read, oppgave.fagsakSaksnummer, oppgave.aktorId, false)
     }
 
-    override fun harTilgangTilOppgaveV3(oppgave: no.nav.k9.los.nyoppgavestyring.visningoguttrekk.Oppgave, bruker: Saksbehandler): Boolean {
-         if (oppgave.hentVerdi("saksnummer") == null) {
-             return true
-         } else return runBlocking {
-             harTilgangTilLesSak(oppgave.hentVerdi("saksnummer")!!, oppgave.hentVerdi("aktorId")!!, bruker)
-         }
+    override suspend fun harTilgangTilOppgaveV3(
+        oppgave: no.nav.k9.los.nyoppgavestyring.visningoguttrekk.Oppgave,
+        action: Action,
+        auditlogging: Boolean
+    ): Boolean {
+        return harTilgang(oppgave.oppgavetype.eksternId, action, oppgave.hentVerdi("saksnummer"), oppgave.hentVerdi("aktorId"), auditlogging)
     }
 
-    override suspend fun harTilgangTilÅReservereOppgave(oppgave: no.nav.k9.los.nyoppgavestyring.visningoguttrekk.Oppgave, bruker: Saksbehandler) : Boolean {
-        val requestBuilder = when (oppgave.oppgavetype.eksternId) {
-            "k9sak" -> {
-                XacmlRequestBuilder()
-                    .addEnvironmentAttribute(ENVIRONMENT_PEP_ID, "srvk9los")
-                    .addResourceAttribute(RESOURCE_DOMENE, DOMENE)
-                    .addResourceAttribute(RESOURCE_TYPE, TILGANG_SAK)
-                    .addActionAttribute(ACTION_ID, "update")
-                    .addResourceAttribute("no.nav.abac.attributter.resource.k9.behandlings_uuid", oppgave.eksternId)  //TODO los skal ikke kjenne til denne detaljen. Oppgavetype må utvides med attributtreferanse
-                    .addAccessSubjectAttribute(SUBJECT_TYPE, INTERNBRUKER)
-                    .addAccessSubjectAttribute(SUBJECTID, bruker.brukerIdent!!)
-            }
-            "k9klage" -> {
-                XacmlRequestBuilder()
+    private suspend fun harTilgang(
+        oppgavetype: String,
+        action: Action,
+        saksnummer: String?,
+        aktorId: String?,
+        auditlogging: Boolean
+    ): Boolean {
+        val identTilInnloggetBruker = azureGraphService.hentIdentTilInnloggetBruker()
+        return when (oppgavetype) {
+            "k9sak", "k9klage" -> {
```

Comment:
Kan ha med k9-tilbake her også, det er samme tilgangskontroll der som for k9sak og k9klage

Reply:
Skulle gjerne hatt enum. Men da må det nesten bli en enum med hvilke oppgavetyper som støttes av PepClient. Siden oppgavetype som entitet er generalisert og hentes fra databasen. Vet ikke om det er så mye å hente da?